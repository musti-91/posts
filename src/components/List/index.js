// @flow
import React from "react";
import Item from "../Item";

type ID = number | string;
type Props = {
  data: *,
  onClick: (id: ID) => void,
  onChecked: (id: ID) => void,
  onDelete: (id: ID) => void
};

const styles = {
  color: "red",
  fontFamily: "Fira Code",
  fontSize: "1em",
  padding: ".4em",
  margin: "0 auto",
  width: "50em"
};
const List = ({ data, onDelete, onChecked, onClick }: Props) => {
  return (
    <div style={styles}>
      {data &&
        typeof data === "object" &&
        data.map(item => (
          <div key={item.id}>
            <Item
              title={item.title}
              id={item.id}
              body={item.body ? item.body : ""}
              checked={item.completed}
              onClick={id => onClick(id)}
              onChecked={id => onChecked(id)}
              onDelete={id => onDelete(id)}
            />
          </div>
        ))}
    </div>
  );
};

export default List;
