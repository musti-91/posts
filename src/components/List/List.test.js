/**
 * @jest-environment jsdom
 */
/* eslint-disable react/display-name */
import React from "react";
import { shallow } from "enzyme";
import MockDate from "mockdate";
/**
 *import  Component
 */
import List from "./index";
/**
 * props
 */
const props = {
  data: [
    {
      title: "something",
      id: 1,
      completed: true
    }
  ],
  onClick: jest.fn(),
  onChecked: jest.fn(),
  onDelete: jest.fn()
};
/**
 * Tests: List Component
 */
describe("List Component", () => {
  beforeAll(() => {
    MockDate.set("2019-01-01T00:00:00Z");
  });
  /**
   * Snapshot:[default]
   */
  test("should match default snapshot", () => {
    expect.assertions(1);

    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
