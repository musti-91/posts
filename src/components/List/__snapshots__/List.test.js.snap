// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component should match default snapshot 1`] = `
<div
  style={
    Object {
      "color": "red",
      "fontFamily": "Fira Code",
      "fontSize": "1em",
      "margin": "0 auto",
      "padding": ".4em",
      "width": "50em",
    }
  }
>
  <div
    key="1"
  >
    <Item
      body=""
      checked={true}
      id={1}
      onChecked={[Function]}
      onClick={[Function]}
      onDelete={[Function]}
      title="something"
    />
  </div>
</div>
`;
